### ISSUE TITLE
Support `${workspaceFolder}` Within Custom Sandbox Template

### ISSUE THREAD

**Issue by han-tyumi** [ocamllabs/vscode-ocaml-platform#1222]
I use Nix to manage my OCaml projects.
I've looked at [how to use this extension with direnv/nix?](https://github.com/ocamllabs/vscode-ocaml-platform/issues/984#top), but unfortunately this does not work for me because the template is run at the root of my system.

It would be nice if `template` supported `${workspaceFolder}` so that we may specify that as the directory to look for a Nix shell configuration.
Another option could be to make sure that this command is started in the workspace folder.

I've also noticed Haskell's IDE will automatically wait and pickup the environment once it's loaded rather than failing hard.
So that would maybe be an even cleaner option if something like that could also be implemented.
This way you could just use the global sandbox in this scenario which would be even simpler.

This is my current workaround:

```json
{
  "ocaml.sandbox": {
    "kind": "custom",
    "template": "nix develop \"${DIRENV_DIR#-}\" --command $prog $args"
  }
}
```

Ideally I could switch to one of the following based on the options I mentioned above.

>```json
>{
>  "ocaml.sandbox": {
>    "kind": "custom",
>    "template": "nix develop \"${workspaceFolder}\" --command $prog $args"
>  }
>}
>```
>With `${workspaceFolder}` support.

> ```json
> {
>   "ocaml.sandbox": {
>     "kind": "custom",
>     "template": "nix develop --command $prog $args"
>   }
> }
> ```
>With the command definitely running in the workspace folder.

> ```json
> {
>   "ocaml.sandbox": {
>     "kind": "global"
>   }
> }
> ```
>With the extension supporting some form of "waiting" for the expected environment to be loaded.

**Comment by mnxn** [comment]
We do not support the `${workspaceFolder}` variable because workspaces may have multiple folders and sandbox configs can come from different places.

You can use either of the following instead:

1. `${workspaceFolder:folderName}`, which will select the folder based on its name. The syntax comes from VS Code's [[Variables scoped per workspace folder](https://code.visualstudio.com/docs/editor/variables-reference#_variables-scoped-per-workspace-folder)](https://code.visualstudio.com/docs/editor/variables-reference#_variables-scoped-per-workspace-folder). 

    (implemented in #424)
    
2. `${firstWorkspaceFolder}`, which will use the first folder in the workspace.

    (implemented in #1004)

**Comment by han-tyumi** [comment]
> We do not support the `${workspaceFolder}` variable because workspaces may have multiple folders and sandbox configs can come from different places.
> 
> You can use either of the following instead:
> 
> 1. `${workspaceFolder:folderName}`, which will select the folder based on its name. The syntax comes from VS Code's [Variables scoped per workspace folder](https://code.visualstudio.com/docs/editor/variables-reference#_variables-scoped-per-workspace-folder).
>    (implemented in [Allow using workspaceFolder placeholder variable for more portable sandbox configurations #424](https://github.com/ocamllabs/vscode-ocaml-platform/pull/424))
> 2. `${firstWorkspaceFolder}`, which will use the first folder in the workspace.
>    (implemented in [Add support for `${firstWorkspaceFolder}` as a placeholder variable in configuration files #1004](https://github.com/ocamllabs/vscode-ocaml-platform/pull/1004))

Oh thanks for pointing that out 👍. Didn't see anything about those in the README / docs.
`${firstWorkspaceFolder}` works for me.

The other options may still be nicer in future.
